piptools/repositories/pypi.py

```diff
@@ -371,10 +383,7 @@ def _get_hashes_from_files(self, ireq: InstallRequirement) -> set[str]:
             ireq.specifier.filter(candidate.version for candidate in all_candidates)
         )
         matching_candidates = candidates_by_version[matching_versions[0]]
-
-        return {
-            self._get_file_hash(candidate.link) for candidate in matching_candidates
-        }
+        return matching_candidates
```

Comment:
```suggestion
        return candidates_by_version[matching_versions[0]]
```

I thinks there's no need for intermediate variable.

Reply:
Updated :+1: